scripts/o3de/tests/test_export_project.py

```diff
@@ -0,0 +1,152 @@
+    # TEST_PYTHON_SCRIPT
+    # successful cases
+    pytest.param(TEST_PYTHON_SCRIPT, ['456'], False, pathlib.PurePath("test_project"), pathlib.PurePath("test_project/ExportScripts"), pathlib.PurePath("test_output.txt"), False, check_for_o3de_context_arg, 0),
+    pytest.param(TEST_PYTHON_SCRIPT, ['456'], True, pathlib.PurePath("test_project"), pathlib.PurePath("test_project/ExportScripts"), pathlib.PurePath("test_output.txt"), False, check_for_o3de_context_arg, 0),
+    pytest.param(TEST_PYTHON_SCRIPT, ['456'], True, pathlib.PurePath("test_project"), pathlib.PurePath("export_scripts"), pathlib.PurePath("test_output.txt"), False, check_for_o3de_context_arg, 0),
+    pytest.param(TEST_PYTHON_SCRIPT, [456], True, pathlib.PurePath("test_project"), pathlib.PurePath("export_scripts"), pathlib.PurePath("test_output.txt"), False, check_for_o3de_context_arg, 0),
+    # failure cases
+    pytest.param(TEST_PYTHON_SCRIPT, [], True, pathlib.PurePath("test_project"), pathlib.PurePath("export_scripts"), pathlib.PurePath("test_output.txt"), False, check_for_o3de_context_arg, 1),
+    pytest.param(TEST_PYTHON_SCRIPT, [456], False, pathlib.PurePath("test_project"), pathlib.PurePath("export_scripts"), pathlib.PurePath("test_output.txt"), False, check_for_o3de_context_arg, 1),
+    # TEST_ERR_PYTHON_SCRIPT
+    pytest.param(TEST_ERR_PYTHON_SCRIPT, [], True, pathlib.PurePath("test_project"), pathlib.PurePath("export_scripts"), None, True, None, 1)
+])
+def test_export_script(tmp_path,
+                       input_script, 
+                       args, 
+                       should_pass_project_folder, 
+                       project_folder_subpath, 
+                       script_folder_subpath, 
+                       output_filename, 
+                       is_expecting_error, 
+                       output_evaluation_func,
+                       expected_result):
+    import sys
+
+    project_folder = tmp_path / project_folder_subpath
+    project_folder.mkdir()
+
+    script_folder = tmp_path / script_folder_subpath
+    script_folder.mkdir()
+
+
+    project_json = project_folder / "project.json"
+    project_json.write_text(TEST_PROJECT_JSON_PAYLOAD)
+
+
+    test_script = script_folder / "test.py"
+    test_script.write_text(input_script)
+
+    if output_filename:
+        test_output = script_folder / output_filename
+
+        assert not test_output.is_file()
+    
+    result = _export_script(test_script, project_folder if should_pass_project_folder else None, args)
+
+    assert result == expected_result
+
+
+    # only check for these if we're simulating a successful case
+    if result == 0 and not is_expecting_error:
+        assert str(script_folder) in sys.path
+
+        if output_filename:
+            assert test_output.is_file()
+
+            with open(test_output, 'r') as t_out:
```

Comment:
nitpick: `test_output.open('r')` can be used here for consistency with the `is_file` call check.
```suggestion
            assert test_output.is_file()

            with test_output.open('r') as t_out:
```